### PR TITLE
[PREVIEW] - Feature/sl 1136/amend listing request

### PR DESCRIPTION
**Pull request by radoslawlandowskikainos**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -42,7 +46,7 @@ const routes: Routes = [
 ];
 
 @NgModule({
-    imports: [RouterModule.forRoot(routes, {enableTracing: false})],
+    imports: [RouterModule.forRoot(routes, {enableTracing: true})],
```
Comment (r-kruk-kainos) [inline src/app/app-routing.module.ts]: Should it be here?

File: src/app/core/reference/resolvers/case-types.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@Injectable()
+export class CaseTypesResolver extends BaseResolver implements Resolve<CaseType[]> {
+
+    constructor(private readonly rds: ReferenceDataService) {
+        super()
+    }
+
+    resolve(next: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<CaseType[]> {
+        // return super.getOrFetchData(this.rds, 'fetchCaseTypes', 'getCaseTypes');
```
Comment (r-kruk-kainos) [inline src/app/core/reference/resolvers/case-types.resolver.ts]: comment?

File: src/app/hearing-part/components/listing-update/listing-update.component.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    styleUrls: ['./listing-update.component.scss']
+})
+export class ListingRequestEditComponent {
+    @ViewChild('listingCreateForm') listingFormGroup: FormGroupDirective;
+    @ViewChild('notesComponent') notesComponent: ListingNoteListComponent;
+
+    @Input() set data(value: ListingRequestViewmodel) {
+        this.listing = value;
+
+        if (this.listing.hearing.duration.asHours() >= 24) {
```
Comment (r-kruk-kainos) [inline src/app/hearing-part/components/listing-update/listing-update.component.ts]: This check can be now changed as there is a flag in db is_multisession.

File: src/app/hearing-part/models/listing-create.ts [hunk 4]
```diff
@@ -5,3 +5,16 @@ export interface ListingCreate {
     hearing: UpdateHearingRequest;
     notes: Note[]
 }
+
+export interface ListingRequestViewmodel {
+    hearing: UpdateHearingRequest;
+    notes: Note[]
+}
+
+export function isMultiSessionListing(listing: ListingCreate): boolean {
```
Comment (r-kruk-kainos) [inline src/app/hearing-part/models/listing-create.ts]: Not needed anymore.

File: src/app/hearing-part/components/listing-update/listing-update.component.html [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                                             required>
+                                <mat-radio-button id="single-session-radio" [value]="0" [disabled]="this.isMultiSession" >Single session</mat-radio-button>
+                                <mat-radio-button id="multi-session-radio" [value]="1" [disabled]="this.isMultiSession" >Multi session</mat-radio-button>
+                            </mat-radio-group>
+                        </div>
+
+                        <div formGroupName="listingType">
+                            <div *ngIf="chosenListingType === listingType.Single">
+                                <mat-form-field>
+                                    <input id="duration-minutes" formControlName="duration" [value]="listing.hearing.duration | appDurationAsMinutes" (ngModelChange)="updateDuration($event, 'minute')" type="number" matInput placeholder="Estimated duration *" min="1" max="1439" step="1" >
```
Comment (zielinskikamil) [inline src/app/hearing-part/components/listing-update/listing-update.component.html]: why max="1439" ? :D

**Pull request closed without merging**